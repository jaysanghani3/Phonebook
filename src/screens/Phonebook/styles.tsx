import {StyleSheet} from 'react-native';
import Colors from '../../utils/Colors';

const styles = StyleSheet.create({
  headerContainer: {
    backgroundColor: Colors.appColor,
    padding: 10,
    paddingHorizontal: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  headerBtn: {
    paddingVertical: 7,
    width: 80,
    backgroundColor: Colors.whiteWith20Op,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 50,
  },
  headerText: {
    fontSize: 12,
    color: Colors.white,
  },
  profileImgContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.appColorWith50Op,
    padding: 10,
    height: 250,
    marginBottom: 20,
    borderBottomEndRadius: 60,
    borderBottomStartRadius: 60,
  },
  profileImg: {
    width: 150,
    height: 150,
    borderRadius: 80,
    alignSelf: 'center',
  },
  imgBtnContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 20,
  },
  imgBtn: {
    padding: 10,
    backgroundColor: Colors.whiteWith20Op,
    borderRadius: 50,
    marginHorizontal: 10,
  },
  imgSize: {
    width: 30,
    height: 30,
  },
  btnText: {
    fontSize: 15,
    color: Colors.white,
  },
  inputContainer: {
    marginTop: 5,
    marginHorizontal: 20,
  },
  textArea: {
    borderRadius: 30,
    borderWidth: 1,
    borderColor: Colors.gray,
    backgroundColor: Colors.white,
    paddingStart: 15,
  },
  delContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 20,
  },
  deleteBtn: {
    backgroundColor: Colors.error,
    padding: 10,
    justifyContent: 'center',
    width: 150,
    borderRadius: 30,
    flexDirection: 'row',
    alignItems: 'center',
  },
  delImgSize: {
    width: 20,
    height: 20,
  },
  deleteBtnText: {
    fontSize: 17,
    color: Colors.white,
    marginStart: 20,
  },
  modelContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalView: {
    width: '70%',
    backgroundColor: 'white',
    borderRadius: 20,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  dltTitle: {
    paddingTop: 20,
    padding: 10,
    fontSize: 15,
    fontWeight: 'bold',
  },
  delBtn: {
    backgroundColor: Colors.error,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    width: '50%',
    borderBottomRightRadius: 20,
  },
  delBtnContainer: {
    flexDirection: 'row',
    borderTopWidth: 1,
    borderColor: Colors.blackWith50Op,
  },
  btnWidth: {
    justifyContent: 'center',
    alignItems: 'center',
    width: '50%',
  },
});

export default styles;
