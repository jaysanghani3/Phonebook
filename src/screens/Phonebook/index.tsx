import {
  View,
  Image,
  TouchableOpacity,
  TextInput,
  Text,
  ScrollView,
  Keyboard,
  Modal,
} from 'react-native';
import React, {useState, useEffect} from 'react';
import globalStyles from '../../utils/globalStyles';
import styles from './styles';
import Icons from '../../assets';
import Strings from '../../translations/Strings';
import {InputModeOptions} from 'react-native/Libraries/Components/TextInput/TextInput';
import {launchCamera, launchImageLibrary} from 'react-native-image-picker';
import {
  ImageLibraryOptions,
  CameraOptions,
} from 'react-native-image-picker/src/types';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RouteStackParamsList} from '../../navigators/AppNavigators';
import {Routes} from '../../navigators/routes';
import {
  createContact,
  updateContact,
  deleteContact,
  createContactOffline,
  updateContactOffline,
} from '../../services/firestore';
import {isMobileNumberValid, isNameValid} from '../../utils/validations';
import NetInfo from '@react-native-community/netinfo';
import Loader from '../../components/Loader';

type PhonebookProps = NativeStackScreenProps<
  RouteStackParamsList,
  Routes.Phonebook
>;

type IFields = {
  title: string;
  value: string;
  placeholder: string;
  inputType: InputModeOptions;
  errorMsg: string;
}[];
interface IContactDetails {
  firstNameValue: string;
  lastNameValue: string;
  phoneNumberValue: string;
  mobileNumberValue: string;
  workNumberValue: string;
  homeNumberValue: string;
  imageValue: string | undefined;
}
interface IError {
  firstNameValue?: string;
  lastNameValue?: string;
  phoneNumberValue?: string;
  mobileNumberValue?: string;
  workNumberValue?: string;
  homeNumberValue?: string;
}
const Phonebook = ({navigation, route}: PhonebookProps) => {
  const {
    id,
    firstNameValue,
    lastNameValue,
    phoneNumberValue,
    mobileNumberValue,
    workNumberValue,
    homeNumberValue,
    imageValue,
  } = route.params;

  const [contactDetails, setContactDetails] = useState<IContactDetails>({
    firstNameValue: firstNameValue || '',
    lastNameValue: lastNameValue || '',
    phoneNumberValue: phoneNumberValue || '',
    mobileNumberValue: mobileNumberValue || '',
    workNumberValue: workNumberValue || '',
    homeNumberValue: homeNumberValue || '',
    imageValue: imageValue || '',
  });
  const Fields: IFields = [
    {
      title: 'firstNameValue',
      value: contactDetails.firstNameValue,
      placeholder: Strings.fnamePlaceholder,
      inputType: 'text',
      errorMsg: Strings.required,
    },
    {
      title: 'lastNameValue',
      value: contactDetails.lastNameValue,
      placeholder: Strings.lnamePlaceholder,
      inputType: 'text',
      errorMsg: Strings.required,
    },
    {
      title: 'phoneNumberValue',
      value: contactDetails.phoneNumberValue,
      placeholder: Strings.phonePlaceholder,
      inputType: 'tel',
      errorMsg: Strings.required,
    },
    {
      title: 'mobileNumberValue',
      value: contactDetails.mobileNumberValue,
      placeholder: Strings.mobilePlaceholder,
      inputType: 'tel',
      errorMsg: Strings.required,
    },
    {
      title: 'workNumberValue',
      value: contactDetails.workNumberValue,
      placeholder: Strings.workPlaceholder,
      inputType: 'tel',
      errorMsg: Strings.required,
    },
    {
      title: 'homeNumberValue',
      value: contactDetails.homeNumberValue,
      placeholder: Strings.homePlaceholder,
      inputType: 'tel',
      errorMsg: Strings.required,
    },
  ];
  const [error, setError] = useState<IError>({
    firstNameValue: '',
    lastNameValue: '',
    phoneNumberValue: '',
    mobileNumberValue: '',
    workNumberValue: '',
    homeNumberValue: '',
  });
  const [isConnected, setIsConnected] = useState<boolean>(false);
  const [loading, setLoading] = useState<boolean>(false);
  const [modalVisible, setModalVisible] = useState<boolean>(false);
  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener(state => {
      return setIsConnected(state.isConnected ?? false);
    });
    return () => {
      unsubscribe();
    };
  }, []);
  const handleCancel = () => {
    navigation.goBack();
  };

  const handleAdd = async () => {
    Keyboard.dismiss();
    if (!contactDetails.firstNameValue || !contactDetails.phoneNumberValue) {
      setError({
        ...error,
        firstNameValue: !contactDetails.firstNameValue ? Strings.required : '',
        phoneNumberValue: !contactDetails.phoneNumberValue
          ? Strings.required
          : '',
      });
      return;
    }
    if (!isNameValid(contactDetails.firstNameValue)) {
      setError({
        firstNameValue: Strings.invalidName,
      });
      return;
    }
    if (contactDetails.lastNameValue) {
      if (!isNameValid(contactDetails.lastNameValue)) {
        setError({
          lastNameValue: Strings.invalidName,
        });
        return;
      }
    }
    if (!isMobileNumberValid(contactDetails.phoneNumberValue)) {
      setError({phoneNumberValue: Strings.invalidPhone});
      return;
    }
    if (contactDetails.mobileNumberValue) {
      if (!isMobileNumberValid(contactDetails.mobileNumberValue)) {
        setError({mobileNumberValue: Strings.invalidPhone});
        return;
      }
    }
    if (contactDetails.workNumberValue) {
      if (!isMobileNumberValid(contactDetails.workNumberValue)) {
        setError({workNumberValue: Strings.invalidPhone});
        return;
      }
    }
    if (contactDetails.homeNumberValue) {
      if (!isMobileNumberValid(contactDetails.homeNumberValue)) {
        setError({homeNumberValue: Strings.invalidPhone});
        return;
      }
    }

    setError({
      firstNameValue: '',
      lastNameValue: '',
      phoneNumberValue: '',
      mobileNumberValue: '',
      workNumberValue: '',
      homeNumberValue: '',
    });
    setLoading(true);
    if (!id) {
      if (!isConnected) {
        createContactOffline(contactDetails, contactDetails.imageValue ?? '');
      } else {
        await createContact(contactDetails, contactDetails.imageValue ?? '');
      }
    } else {
      if (!isConnected) {
        updateContactOffline(
          id,
          contactDetails,
          contactDetails.imageValue ?? '',
        );
      } else {
        updateContact(id, contactDetails, contactDetails.imageValue ?? '');
      }
    }
    navigation.goBack();
  };

  const handleGallery = () => {
    const options: ImageLibraryOptions = {
      mediaType: 'photo',
      quality: 1,
      includeBase64: true,
    };
    launchImageLibrary(options, response => {
      if (response.didCancel) {
        console.log('User cancelled image picker');
      } else if (response.errorCode) {
        console.log('ImagePicker Error: ', response.errorMessage);
      } else {
        let imageUri = response.assets?.[0]?.uri;
        console.log('imageUri', imageUri);
        setContactDetails({...contactDetails, imageValue: imageUri});
      }
    });
  };

  const handleCamera = async () => {
    const options: CameraOptions = {
      mediaType: 'photo',
      includeBase64: false,
      maxHeight: 2000,
      maxWidth: 2000,
      saveToPhotos: true,
    };

    const result = await launchCamera(options);
    const imageUri = result.assets?.[0]?.uri;
    setContactDetails({...contactDetails, imageValue: imageUri});
  };

  const handleDelete = () => {
    setModalVisible(!modalVisible);
    deleteContact(id);
    navigation.goBack();
  };

  return (
    <View style={globalStyles.flex1}>
      <Loader state={loading} />
      <View style={styles.headerContainer}>
        <TouchableOpacity style={styles.headerBtn} onPress={handleCancel}>
          <Text style={styles.headerText}>{Strings.cancelBtn}</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.headerBtn} onPress={handleAdd}>
          <Text style={styles.headerText}>
            {!id ? Strings.add : Strings.update}
          </Text>
        </TouchableOpacity>
      </View>
      <ScrollView>
        <View style={styles.profileImgContainer}>
          <Image
            source={
              contactDetails.imageValue
                ? {uri: contactDetails?.imageValue}
                : Icons.profile
            }
            style={styles.profileImg}
          />
          <View style={styles.imgBtnContainer}>
            <TouchableOpacity onPress={handleGallery} style={styles.imgBtn}>
              <Image source={Icons.gallery} style={styles.imgSize} />
            </TouchableOpacity>
            <TouchableOpacity onPress={handleCamera} style={styles.imgBtn}>
              <Image source={Icons.camera} style={styles.imgSize} />
            </TouchableOpacity>
          </View>
        </View>
        {Fields.map((field, index) => (
          <View key={index} style={styles.inputContainer}>
            <TextInput
              value={field.value}
              onChangeText={newValue => {
                setContactDetails({...contactDetails, [field.title]: newValue});
              }}
              style={styles.textArea}
              placeholder={field.placeholder}
              inputMode={field.inputType}
            />
            <Text style={globalStyles.error}>
              {error[field.title as keyof typeof error]}
            </Text>
          </View>
        ))}
        {id && (
          <View style={styles.delContainer}>
            <TouchableOpacity
              style={styles.deleteBtn}
              onPress={() => setModalVisible(true)}>
              <Image source={Icons.delete} style={styles.delImgSize} />
              <Text style={styles.deleteBtnText}>{Strings.delete}</Text>
            </TouchableOpacity>
          </View>
        )}

        <Modal
          animationType="slide"
          transparent={true}
          visible={modalVisible}
          onRequestClose={() => {
            setModalVisible(!modalVisible);
          }}>
          <View style={styles.modelContainer}>
            <View style={styles.modalView}>
              <Text style={styles.dltTitle}>{Strings.dltMsg}</Text>
              <View style={styles.delBtnContainer}>
                <TouchableOpacity
                  onPress={() => setModalVisible(!modalVisible)}
                  style={styles.btnWidth}>
                  <Text>{Strings.cancel}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={handleDelete} style={styles.delBtn}>
                  <Text style={styles.btnText}>{Strings.yes}</Text>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        </Modal>
      </ScrollView>
    </View>
  );
};

export default Phonebook;
