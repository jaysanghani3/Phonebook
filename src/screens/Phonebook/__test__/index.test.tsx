import renderer from 'react-test-renderer';
import Phonebook from '../index';

it('renders correctly', () => {
  const mockRoute = {
    params: {
      id: 'test',
      firstNameValue: 'test',
      lastNameValue: 'test',
      phoneNumberValue: 'test',
      mobileNumberValue: 'test',
      workNumberValue: 'test',
      homeNumberValue: 'test',
      imageValue: 'test',
    },
  };
  const tree = renderer
    .create(<Phonebook navigation={undefined} route={mockRoute} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
