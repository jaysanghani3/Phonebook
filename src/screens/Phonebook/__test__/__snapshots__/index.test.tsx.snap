// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#66347f",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "borderRadius": 50,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 7,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          }
        }
      >
        Cancel
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "borderRadius": 50,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 7,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          }
        }
      >
        Update
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(102, 52, 127, 0.5)",
            "borderBottomEndRadius": 60,
            "borderBottomStartRadius": 60,
            "height": 250,
            "justifyContent": "center",
            "marginBottom": 20,
            "padding": 10,
          }
        }
      >
        <Image
          source={
            {
              "uri": "test",
            }
          }
          style={
            {
              "alignSelf": "center",
              "borderRadius": 80,
              "height": 150,
              "width": 150,
            }
          }
        />
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderRadius": 50,
                "marginHorizontal": 10,
                "opacity": 1,
                "padding": 10,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/gallery.png",
                }
              }
              style={
                {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderRadius": 50,
                "marginHorizontal": 10,
                "opacity": 1,
                "padding": 10,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/camera.png",
                }
              }
              style={
                {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginHorizontal": 20,
            "marginTop": 5,
          }
        }
      >
        <TextInput
          inputMode="text"
          onChangeText={[Function]}
          placeholder="First Name"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8D8D8D",
              "borderRadius": 30,
              "borderWidth": 1,
              "paddingStart": 15,
            }
          }
          value="test"
        />
        <Text
          style={
            {
              "color": "#FF0000",
              "fontSize": 12,
              "marginStart": 10,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginHorizontal": 20,
            "marginTop": 5,
          }
        }
      >
        <TextInput
          inputMode="text"
          onChangeText={[Function]}
          placeholder="Last Name"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8D8D8D",
              "borderRadius": 30,
              "borderWidth": 1,
              "paddingStart": 15,
            }
          }
          value="test"
        />
        <Text
          style={
            {
              "color": "#FF0000",
              "fontSize": 12,
              "marginStart": 10,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginHorizontal": 20,
            "marginTop": 5,
          }
        }
      >
        <TextInput
          inputMode="tel"
          onChangeText={[Function]}
          placeholder="Phone"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8D8D8D",
              "borderRadius": 30,
              "borderWidth": 1,
              "paddingStart": 15,
            }
          }
          value="test"
        />
        <Text
          style={
            {
              "color": "#FF0000",
              "fontSize": 12,
              "marginStart": 10,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginHorizontal": 20,
            "marginTop": 5,
          }
        }
      >
        <TextInput
          inputMode="tel"
          onChangeText={[Function]}
          placeholder="Mobile"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8D8D8D",
              "borderRadius": 30,
              "borderWidth": 1,
              "paddingStart": 15,
            }
          }
          value="test"
        />
        <Text
          style={
            {
              "color": "#FF0000",
              "fontSize": 12,
              "marginStart": 10,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginHorizontal": 20,
            "marginTop": 5,
          }
        }
      >
        <TextInput
          inputMode="tel"
          onChangeText={[Function]}
          placeholder="Work"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8D8D8D",
              "borderRadius": 30,
              "borderWidth": 1,
              "paddingStart": 15,
            }
          }
          value="test"
        />
        <Text
          style={
            {
              "color": "#FF0000",
              "fontSize": 12,
              "marginStart": 10,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginHorizontal": 20,
            "marginTop": 5,
          }
        }
      >
        <TextInput
          inputMode="tel"
          onChangeText={[Function]}
          placeholder="Home"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8D8D8D",
              "borderRadius": 30,
              "borderWidth": 1,
              "paddingStart": 15,
            }
          }
          value="test"
        />
        <Text
          style={
            {
              "color": "#FF0000",
              "fontSize": 12,
              "marginStart": 10,
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FF0000",
              "borderRadius": 30,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
              "width": 150,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/delete.png",
              }
            }
            style={
              {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 17,
                "marginStart": 20,
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
